src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -818,8 +818,8 @@ instruct reduce_andI(iRegINoSp dst, iRegIorL2I src1, vReg src2, vReg tmp) %{
             "vmv.x.s $dst, $tmp" %}
   ins_encode %{
     BasicType bt = Matcher::vector_element_basic_type(this, $src2);
-    __ reduce_operation($dst$$Register, as_VectorRegister($tmp$$reg),
```

Comment:
Given the predicate `Matcher::vector_element_basic_type(n->in(2)) != T_LONG`, is there a risk it's going to match for types that don't fit the assert at https://github.com/openjdk/jdk/pull/11036/files#diff-35eb1d2f1e2f0514dd46bd7fbad49ff2c87703d5a3041a6433956df00a3fe6e6R1696 ?

It seems the same approach as `reduce_addI` should be taken with the following predicate:
```
  predicate(Matcher::vector_element_basic_type(n->in(2)) == T_BYTE || 
            Matcher::vector_element_basic_type(n->in(2)) == T_SHORT ||
            Matcher::vector_element_basic_type(n->in(2)) == T_INT);
```

Reply:
> Given the predicate `Matcher::vector_element_basic_type(n->in(2)) != T_LONG`, is there a risk it's going to match for types that don't fit the assert at https://github.com/openjdk/jdk/pull/11036/files#diff-35eb1d2f1e2f0514dd46bd7fbad49ff2c87703d5a3041a6433956df00a3fe6e6R1696 ?
> 
> It seems the same approach as `reduce_addI` should be taken with the following predicate:
> 
> ```
>   predicate(Matcher::vector_element_basic_type(n->in(2)) == T_BYTE || 
>             Matcher::vector_element_basic_type(n->in(2)) == T_SHORT ||
>             Matcher::vector_element_basic_type(n->in(2)) == T_INT);
> ```

Thanks, Referring to aarch64, currently AndReductionV, OrReductionV, XorReductionV will only match T_BYTE, T_SHORT, T_INT, T_LONG types, and there should be no such risk at present. But for the sake of this uncertain risk, I will modify it later to use the same predicate as reduce_addI.

Reply:
> Given the predicate `Matcher::vector_element_basic_type(n->in(2)) != T_LONG`, is there a risk it's going to match for types that don't fit the assert at https://github.com/openjdk/jdk/pull/11036/files#diff-35eb1d2f1e2f0514dd46bd7fbad49ff2c87703d5a3041a6433956df00a3fe6e6R1696 ?
> 
> It seems the same approach as `reduce_addI` should be taken with the following predicate:
> 
> ```
>   predicate(Matcher::vector_element_basic_type(n->in(2)) == T_BYTE || 
>             Matcher::vector_element_basic_type(n->in(2)) == T_SHORT ||
>             Matcher::vector_element_basic_type(n->in(2)) == T_INT);
> ```

Thanks. referring to aarch64 before, for unnecessary risks, now the same predicate is used as reduce_addI.